flang/include/flang/Parser/parse-tree.h

```diff
@@ -2236,16 +2243,29 @@ struct ConcurrentHeader {
       t;
 };
 
+// F'2023 R1131 reduce-operation ->
+//                + | * | .AND. | .OR. | .EQV. | .NEQV. |
+//                MAX | MIN | IAND | IOR | IEOR
+struct ReduceOperation {
+  UNION_CLASS_BOILERPLATE(ReduceOperation);
+  std::variant<DefinedOperator, ProcedureDesignator> u;
```

Comment:
I'll repeat a comment sent personally to you before.

>I tried an implementation in a similar way to AccReductionOperator::Operator. It cannot tell the reason for parsing failures. The name of AccReductionOperator::Operator is redundant. I tried to add ReduceOperation::Add directly (without ::Operator::), but the OpenACC version has a sourced parser. Defining 'source' in 'ReduceOperation' does not match with other code.

There was no CUF REDUCE support that time. Should I check it?

Reply:
Yes.  And you can handle the error recovery of an enumerated operator in the parser rather than allowing the parse tree to hold an overly general procedure designator.